File: packages/icoder-sdk/src/resources/platform.ts
import type { AxiosInstance } from 'axios';
import { requestConfig, type iCoDerRequestOptions } from '../request-options.js';

export interface EnvironmentPlanRequest {
  environment_code: 'eu' | 'us' | 'cn';
  region_code: string;
  tenant_id?: string;
  dry_run?: boolean;
}

export class PlatformResource {
  constructor(private readonly http: AxiosInstance) {}

  async listEnvironments(options?: iCoDerRequestOptions): Promise<Record<string, unknown>> {
    const { data } = await this.http.get(
      '/api/platform/environments', requestConfig(options),
    );
    return data;
  }

  async listRegions(options?: iCoDerRequestOptions): Promise<Record<string, unknown>> {
    const { data } = await this.http.get('/api/platform/regions', requestConfig(options));
    return data;
  }

  async planEnvironment(
    request: EnvironmentPlanRequest,
    options?: iCoDerRequestOptions,
  ): Promise<Record<string, unknown>> {
    const { data } = await this.http.post('/api/platform/environments', {
      ...request,
      dry_run: request.dry_run ?? true,
    }, requestConfig(options));
    return data;
  }

  async currentTenant(options?: iCoDerRequestOptions): Promise<Record<string, unknown>> {
    const { data } = await this.http.get('/api/tenants/current', requestConfig(options));
    return data;
  }

  async tenantEnvironments(
    tenantId: string,
    options?: iCoDerRequestOptions,
  ): Promise<Record<string, unknown>> {
    const { data } = await this.http.get(
      `/api/tenants/${encodeURIComponent(tenantId)}/environments`,
      requestConfig(options),
    );
    return data;
  }
}

File: packages/icoder-python/icoder_sdk/resources/platform.py
"""Declarative Environments/Regions and Tenant compatibility resources."""

from __future__ import annotations

from typing import Any
from urllib.parse import quote

from ..client import iCoDerClient
from ..request_options import RequestOptions


class PlatformResource:
    def __init__(self, client: iCoDerClient):
        self._client = client

    def list_environments(
        self, request_options: RequestOptions | None = None,
    ) -> dict[str, Any]:
        response = self._client.get(
            "/api/platform/environments", request_options=request_options,
        )
        response.raise_for_status()
        return response.json()

    def list_regions(
        self, request_options: RequestOptions | None = None,
    ) -> dict[str, Any]:
        response = self._client.get(
            "/api/platform/regions", request_options=request_options,
        )
        response.raise_for_status()
        return response.json()

    def plan_environment(
        self,
        environment_code: str,
        region_code: str,
        *,
        tenant_id: str | None = None,
        request_options: RequestOptions | None = None,
    ) -> dict[str, Any]:
        response = self._client.post(
            "/api/platform/environments",
            json={
                "environment_code": environment_code,
                "region_code": region_code,
                "tenant_id": tenant_id,
                "dry_run": True,
            },
            request_options=request_options,
        )
        response.raise_for_status()
        return response.json()

    def current_tenant(
        self, request_options: RequestOptions | None = None,
    ) -> dict[str, Any]:
        response = self._client.get(
            "/api/tenants/current", request_options=request_options,
        )
        response.raise_for_status()
        return response.json()

    def tenant_environments(
        self,
        tenant_id: str,
        request_options: RequestOptions | None = None,
    ) -> dict[str, Any]:
        response = self._client.get(
            f"/api/tenants/{quote(tenant_id, safe='')}/environments",
            request_options=request_options,
        )
        response.raise_for_status()
        return response.json()
